File: Greedy/42885.py
def solution(people: list, limit: int) -> int:
    answer = 0
    people.sort()
    
    left = 0
    right = len(people) - 1
    
    while left < right:
        if people[left] + people[right] <= limit:
            left += 1
            right -= 1
        else:
            right -= 1
        answer += 1
    
    if left == right: 
    # Testcase 2 - 값이 다 더해지지 않을 시, left=right 같아짐.
        answer += 1
    
    return answer


solution([70, 50, 80, 50], 100)

File: Javascript/Greedy/42885.js
function solution(people, limit) {
    var answer = 0;
    
    // 오름차순 정렬
    people.sort((a, b) => a - b);
    
    // 시작 Flag
    let left = 0;
    let right = people.length - 1;
    
    while(left < right){
        if (limit >= people[left] + people[right]){
            left += 1;
            right -= 1;
            answer += 1;
        }
        else{
            right -= 1;
            answer += 1;
        }
    }
    if (left == right){
        answer += 1;
    }
    
    
    return answer;
}